src/server/conn.rs

```diff
@@ -231,6 +236,12 @@ pub struct Parts<T, S> {
 impl Http {
     /// Creates a new instance of the HTTP protocol, ready to spawn a server or
     /// start accepting connections.
+    #[cfg_attr(
```

Comment:
I'd put these attributes on the `struct`s instead. I know it explodes some warnings in here, but you can add `#[cfg_attr(feature = "deprecated", allow(deprecated))]` on the complaining `impl Whatever` blocks. You can see what I did in #3156.

Oh, also, can we make the deprecations a separate PR?

Reply:
Sure thing!